civis/ml/tests/test_model.py

```diff
@@ -666,50 +690,91 @@ def test_metrics_prediction(mock_file_id_from_run_output):
                                                     client=mock.ANY)
 
 
+###############################################
+# Tests of utilities for CivisML template IDs #
+###############################################
+@pytest.mark.parametrize(
+    'alias, expected_job_type, expected_version',
+    [
+        ('civis-civisml-training', 'training', None),
+        ('civis-civisml-training-v2-3', 'training', 'v2.3'),
+        ('civis-civisml-training-dev', 'training', 'dev'),
+        ('civis-civisml-training-foo-bar', 'training', 'foo-bar'),
+    ],
+)
+def test__get_job_type_version(alias, expected_job_type, expected_version):
+    actual_job_type, actual_version = _model._get_job_type_version(alias)
+    assert actual_job_type == expected_job_type
+    assert actual_version == expected_version
+
+
+@pytest.mark.parametrize(
+    'invalid_alias',
+    ['foobar', 'civis-civisml', 'civis-civisml-', 'civis-civisml-training-',
+     'civis-civisml-training-foobar-']
+)
+def test__get_job_type_version_invalid_alias(invalid_alias):
+    with pytest.raises(ValueError):
+        _model._get_job_type_version(invalid_alias)
+
+
+def test__get_template_ids_all_versions():
+    m_client = create_client_mock()
+    m_client.aliases.list.return_value = TEST_TEMPLATE_ID_ALIAS_OBJECTS
+    actual_template_ids = _model._get_template_ids_all_versions(m_client)
+    expected_template_ids = TEST_TEMPLATE_IDS
+    assert actual_template_ids == expected_template_ids
+
+
+@mock.patch.object(_model, '_get_template_ids_all_versions',
+                   mock.Mock(return_value=TEST_TEMPLATE_IDS))
```

Comment:
Comment: You'll get the same effect by supplying `return_value` directly. Extra keyword arguments get passed directly to the `Mock` object's `__init__` (which can cause subtle bugs if you misspell a keyword parameter in your patch).
```suggestion
                   return_value=TEST_TEMPLATE_IDS)
```

Reply:
I had tried the kwarg `return_value=TEST_TEMPLATE_IDS` but the tests didn't pass, which is why I'm using the supposedly equivalent `mock.Mock(return_value=TEST_TEMPLATE_IDS)`. I was puzzled why `return_value=TEST_TEMPLATE_IDS` didn't work, but wasn't interested enough to spend more time on this.